src/lexer.coffee

```diff
@@ -740,6 +740,12 @@ exports.Lexer = class Lexer
         prev[0] = 'COMPOUND_ASSIGN'
         prev[1] += '='
         prev.data.original += '=' if prev.data?.original
+        prev[2].range = [
```

Comment:
The existing logic failed to expand the location data of the existing eg `or` token when appending the `=` to it (in the case of `or=`)